EtherEncLib.cpp

```diff
@@ -333,12 +339,23 @@ char *EtherEncLib::getParams(void)
     resposta = "Pragma: no-cache\r\n\r\n";
 	m_stack.write(resposta, lenData);
 	m_stack.send();
-
-	for (unsigned i = 0; i < BUFFER_PARAMS_LEN; i++)  m_httpData[i] = tmpData[i];
-
-    return &m_httpData[0];
+*/
+	if ( j > 0 ) {
+		for (unsigned i = 0; i < BUFFER_PARAMS_LEN; i++)  m_httpData[i] = tmpData[i];
+	}
+	return 1;
 }
 
+void EtherEncLib::printHeader(char *respondType, unsigned char dataLen) {
+	char *resposta = (char *)malloc(dataLen);
```

Comment:
Sorry SKA... no malloc() -- reference post: http://arduino.stackexchange.com/questions/682/is-using-malloc-and-free-a-really-bad-idea-on-arduino -- Malloc and Dealloc both cause gaps at Heap area. We need to move that malloc to constructor or we can work with a local buffer. The local buffer is prefered than static buffer (last one is always there) and local bufer will only be stacked at calling time. 

We do have some work to do about these buffers... I think we got 3 of them. One at sketch, one at print function an that one at printHead function...

This printscreen below is my Arduino hanging because of memory blow. That occurs at HelloWorld...

Working on it.

See ya! And Thanks for your contribution! It is amazing work!
Renato

![enc28j60_2](https://cloud.githubusercontent.com/assets/4915969/7753452/e9be41a6-ffbb-11e4-8285-631aad9ec67f.jpg)